library/sha256.c

```diff
@@ -317,47 +313,49 @@ int mbedtls_sha256_update_ret( mbedtls_sha256_context *ctx,
     SHA256_VALIDATE_RET( ctx != NULL );
     SHA256_VALIDATE_RET( ilen == 0 || input != NULL );
 
-    if( ilen == 0 )
+    if( ilen_dup == 0 )
```

Comment:
I think it would be nice to add some comment about `ilen_dup` is used because of `volatile` and FI

Reply:
Added.

Reply:
Thanks